lab-adam/route/vehicle-router.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Router = require('express').Router;
+const debug = require('debug')('vehicle:vehicle-router');
+const jsonParser = require('body-parser').json();
+const Vehicle = require('../model/vehicle.js');
+const storage = require('../lib/storage.js');
+
+const noteRouter = new Router();
+
+noteRouter.get('/', function(req, res, next){
+  debug('hit route GET /api/vehicle');
+  res.json({msg: 'hello'});
+  next();
+});
+
+noteRouter.get('/api/vehicle/:id', function(req, res, next){
+  debug('hit route GET /api/vehicle');
+  if (req.params.id) {
+    Vehicle.fetchVehicle(req.params.id)
+    .then(vehicle => res.json(vehicle))
+    .catch(err => next(err));
+  } else {
```

Comment:
This won't work, because this route will only be triggered if there is something following /api/vehicle in the pathname. (/:id). Therefore, a request to /api/vehicle will not hit this route and will instead receive a 404 response. You'll need to abstract this into its own route, '/api/vehicle.'